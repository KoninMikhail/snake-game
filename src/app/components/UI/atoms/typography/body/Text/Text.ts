import styled from 'styled-components';
import {up} from 'styled-breakpoints';
import {ITextProps} from '@ui/atoms/typography/Typography.interface';

/**
 * @name Text
 * @author Mikhail Konin <dev.konin@gmail.com>
 *
 * Use it for body text
 *
 * @param as - custom element tag
 * @param align - custom alignment
 * @param weight - custom weight
 * @returns {JSX.Element}
 */
export const Text = styled.span<ITextProps>`
    color: ${({ theme }) => theme.colors.body};
    font-size: ${({ theme }) => theme.typography.fontSize.body.mobile};
    font-family: ${({ theme }) => theme.typography.fontFamily.body};
    font-weight: ${({ weight }) => weight ?? 'normal'};

    ${({ align }) => align && `text-align: ${align};`}
    ${up('md')} {
        font-size: ${({ theme }) => theme.typography.fontSize.body.tablet};
    }

    ${up('xl')} {
        font-size: ${({ theme }) => theme.typography.fontSize.body.desktop};
    }
`;
