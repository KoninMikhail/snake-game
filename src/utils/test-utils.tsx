import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
