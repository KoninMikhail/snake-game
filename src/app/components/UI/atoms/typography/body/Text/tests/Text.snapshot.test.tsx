import {Text} from '@ui/atoms/typography/body/Text/Text';
import {renderWithProviders} from '@utils/test-utils';

/**
 * Test
 */
describe('Text', () => {
    it('equal to snapshot', () => {
        const { container } = renderWithProviders(<Text align={'center'}>Arriva</Text>);

        expect(container).toMatchSnapshot();
    });
});
