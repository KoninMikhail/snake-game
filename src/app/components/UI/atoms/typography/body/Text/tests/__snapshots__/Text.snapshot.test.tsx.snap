// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text equal to snapshot 1`] = `
.c0 {
  color: white;
  font-size: 14px;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    font-size: 16px;
  }
}

@media (min-width:1200px) {
  .c0 {
    font-size: 1.1vw;
  }
}

<div>
  <span
    class="c0"
  >
    Arriva
  </span>
</div>
`;
